src/esp/assets/managers/StageAttributesManager.cpp

```diff
@@ -34,42 +34,49 @@ StageAttributesManager::StageAttributesManager(
 }  // StageAttributesManager ctor
 
 StageAttributes::ptr StageAttributesManager::createAttributesTemplate(
-    const std::string& stageAttributesHandle,
+    const std::string& attributesTemplateHandle,
     bool registerTemplate) {
   StageAttributes::ptr attrs;
   std::string msg;
-  std::string strHandle = Cr::Utility::String::lowercase(stageAttributesHandle);
-  bool fileExists = (this->isValidFileName(stageAttributesHandle));
-  if (objectAttributesMgr_->isValidPrimitiveAttributes(stageAttributesHandle)) {
-    // if stageAttributesHandle == some existing primitive attributes, then
+  if (objectAttributesMgr_->isValidPrimitiveAttributes(
+          attributesTemplateHandle)) {
+    // if attributesTemplateHandle == some existing primitive attributes, then
     // this is a primitive-based stage (i.e. a plane) we are building
-    attrs = createPrimBasedAttributesTemplate(stageAttributesHandle,
+    attrs = createPrimBasedAttributesTemplate(attributesTemplateHandle,
                                               registerTemplate);
-    msg = "Primitive Asset (" + stageAttributesHandle + ") Based";
+    msg = "Primitive Asset (" + attributesTemplateHandle + ") Based";
 
-  } else if (fileExists) {
-    if ((strHandle.find("stage_config.json") != std::string::npos) &&
-        fileExists) {
+  } else {
+    std::string JSONTypeExt("stage_config.json");
+    std::string strHandle =
+        Cr::Utility::String::lowercase(attributesTemplateHandle);
+    std::string jsonAttrFileName =
+        (strHandle.find(JSONTypeExt) != std::string::npos)
+            ? std::string(attributesTemplateHandle)
+            : io::changeExtension(attributesTemplateHandle, JSONTypeExt);
+    bool fileExists = (this->isValidFileName(attributesTemplateHandle));
+    bool jsonFileExists = (this->isValidFileName(jsonAttrFileName));
+    if (jsonFileExists) {
```

Comment:
Creation may be called with a mesh/asset file name or a configuration json file name.  If called with a mesh/asset file name, a configuration may still exist that we want to use.  This checks for this configuration, and if it exists, loads it, while if it does not exist, then it will create a default attributes and assign the passed handle as render asset handle in attributes.